File: cli/node/src/application/memory/expiration.ts
import { printError } from '../../output/branding.js';
import { exit } from '../../runtime/exit.js';

/**
 * 校验到期日格式及未来日期约束，失败时结束命令。
 * @param value - 到期日期，格式为 YYYY-MM-DD。
 */
export function _validateExpires(value: string): void {
  if (!/^\d{4}-\d{2}-\d{2}$/.test(value)) {
    printError('Invalid date format for --expires. Use YYYY-MM-DD (e.g. 2025-12-31).');
    exit(1);
  }
  if (new Date(value) <= new Date()) {
    printError('--expires date must be in the future.');
    exit(1);
  }
}

File: cli/python/src/memgo_cli/application/memory/expiration.py
"""记忆过期日期校验。"""

from __future__ import annotations

import re
from datetime import date

import typer
from rich.console import Console

from memgo_cli.output.branding import (
    print_error,
)

console = Console()
err_console = Console(stderr=True)


def _validate_expires(value: str) -> None:
    """校验未来的 YYYY-MM-DD 日期，无效时明确退出。"""
    if not re.match(r"^\d{4}-\d{2}-\d{2}$", value):
        print_error(
            err_console,
            "Invalid date format for --expires. Use YYYY-MM-DD (e.g. 2025-12-31).",
            hint=None,
        )
        raise typer.Exit(1)
    if date.fromisoformat(value) <= date.today():
        print_error(err_console, "--expires date must be in the future.", hint=None)
        raise typer.Exit(1)
